src/Dfe.PlanTech.AzureFunctions/Functions/QueueReceiver.cs

```diff
@@ -60,10 +96,13 @@ private async Task ProcessMessage(ServiceBusReceivedMessage message, ServiceBusM
             }
         }
 
-        private async Task<long> UpsertEntityInDatabase(ContentComponentDbEntity entity)
+        private ContentComponentDbEntity? GetExistingDbEntity(ContentComponentDbEntity entity)
         {
-            var existing = _db.Find(entity.GetType(), entity.Id);
+            return _db.Find(entity.GetType(), entity.Id) as ContentComponentDbEntity ?? null;
```

Comment:
There's the possibility that the DB might return a result that _isn't_ a ContentComponentDbEntity, but would then try to cast it and return null as a result.

Could this cause any problems?

Reply:
Not sure, wouldn't this just mean that the entity doesn't exist in the Db yet? That's my intent with it